nixos/modules/services/misc/duckdns.nix

```
@@ -0,0 +1,67 @@
+{ config, pkgs, lib, ... }:
+let
+cfg = config.services.duckdns;
+in
+{
+  options = {
+    services.duckdns = {
+      enable = lib.mkEnableOption (lib.mdDoc "DuckDNS Dynamic DNS Client");
+
+      tokenFile = lib.mkOption {
+        default = null;
+        type = lib.types.str;
```

This supports both Nix paths and strings

```suggestion
        type = lib.types.path;
```